gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStore.java

```diff
@@ -0,0 +1,341 @@
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;/includes.conf<br>
+ *       &emsp;&emsp;&emsp;&emsp;/dataset2<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;/main.conf<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;/child-dataset<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;/main.conf<br>
+ *     </code>
+ *   </blockquote>
+ * </p>
+ *
+ * <p>
+ *   In the above example, the root of the store is {@code /root/my-simple-store/}. The code automatically assumes that
+ *   this folder contains a directory named {@link #CONFIG_STORE_NAME}. In order to access the dataset
+ *   {@code dataset1/child-dataset} using {@link ConfigClient#getConfig(URI)}, the specified {@link URI} should be
+ *   {@code simple-hdfs://[authority]:[port]/root/my-simple-store/dataset1/child-dataset/}. Note this is the fully
+ *   qualified path to the actual {@code main.conf} file on HDFS, with the {@link #CONFIG_STORE_NAME} and the
+ *   {@code version} directories removed.
+ * </p>
+ *
+ * <p>
+ *   All the {@link Config}s for a dataset should be put in the associated {@code main.conf} file, and all the imports
+ *   should be put in the associated {@code includes.conf} file.
+ * </p>
+ *
+ * <p>
+ *   This class is not responsible for deploying configurations from an external source to HDFS, only for reading them.
+ * </p>
+ *
+ * @see {@link SimpleHDFSConfigStoreFactory}
+ */
+public class SimpleHDFSConfigStore implements ConfigStore {
+
+  public static final String SIMPLE_HDFS_SCHEME_PREFIX = "simple-";
+  public static final String CONFIG_STORE_NAME = "_CONFIG_STORE";
+
+  private final FileSystem fs;
+  private final URI physicalStoreRoot;
+
+  /**
+   * Constructs a {@link SimpleHDFSConfigStore} using a given {@link FileSystem} and a {@link URI} that points to the
+   * physical location of the store root.
+   *
+   * @param fs the {@link FileSystem} the {@link ConfigStore} is stored on.
+   * @param physicalStoreRoot the fully qualified {@link URI} of the store root, the {@link URI#getScheme()} of the
+   *                          {@link URI} should match the {@link FileSystem#getScheme()} of the given {@link FileSystem}.
+   */
+  protected SimpleHDFSConfigStore(FileSystem fs, URI physicalStoreRoot) {
+    Preconditions.checkNotNull(fs, "fs cannot be null!");
+    Preconditions.checkNotNull(physicalStoreRoot, "physicalStoreRoot cannot be null!");
+
+    this.fs = fs;
+
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(physicalStoreRoot.getScheme()),
+        "The physicalStoreRoot must have a valid scheme!");
+    Preconditions.checkArgument(physicalStoreRoot.getScheme().equals(fs.getUri().getScheme()),
+        "The scheme of the physicalStoreRoot and the filesystem must match!");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(physicalStoreRoot.getPath()),
+        "The path of the physicalStoreRoot must be valid as it is the root of the store!");
+
+    this.physicalStoreRoot = physicalStoreRoot;
```

Comment:
Say the user does `getConfig("simple-hdfs://mycluster/mystore/myconfig")`, what will the value of `physicalStoreRoot` be? (It seems that it will be `hdfs://mycluster/mystore`?)
What will the return value of `getStoreRoot()` be? Per contract with the config library, it should be `simple-hdfs://mycluster/mystore`, but I don't believe that's what it will return.